### ISSUE TITLE
The module "List of active subscriptions" breaks JDocument object

### ISSUE THREAD

**Issue by ITPrism** [akeeba/akeebasubs#231]
The module "List of active subscriptions" ([mod_aksubslist] (https://github.com/akeeba/akeebasubs/blob/development/modules/site/aksubslist/mod_aksubslist.php#L44)) overrides JDocument object and breaks the design. It is only if the cache is enabled.

How to reproduce the issue?

1. Enable Joomla! cache.
2. Create menu item "All of my subscriptions".
3. Enable and assign the module __mod_aksubslist__ to that menu item.
4. Open the page "All of my subscriptions".

**Comment by nikosdion** [comment]
I don't think what you are saying means what you think it means. It is not possible to replace the JDocument object from a module. Let alone that there is no such code. Or that I'm using this module on my own site.

If you are sure about what you're saying please file a PR with the fix. I'm curious to see what you'll come up with since what you're saying doesn't actually happen...

**Comment by ITPrism** [comment]
I investigated the issue deeper.
Let me show you what I found.

* First breakpoint is the module Akeeba Subscriptions List 
__FOF30\Container\Container::getInstance__
_modules/mod_aksubslist/mod_aksubslist.php_ (line 44) 

* Second breakpoint is __FOF30\Controller\Controller__
_libraries/fof30/Controller/Controller.php_ (line 500) (breakpoint 2)

* Third breakpoint is __JCacheControllerView__
_libraries/joomla/cache/controller/view.php_ (line 62) (breakpoint 3)
On that __line 62__, the system unserializes the data from the cached file.
This variable contains 1 element __styleSheets__ and 4 elements __scripts__.

* Fourth breakpoint is __JCache__
_libraries/joomla/cache/cache.php_ (line 488) (breakpoint 4)
On __line 488__, the system replaces style sheets and scripts of the document with those from cached file, which are fewer.
In my case, I have __9 _styleSheets__ and __13 _scripts__ in the document object. The system replaces them with the data from the cache which are __4 scripts__ and __1 styleSheet__.

It seems, the problem comes when I use a template of JoomlArt. JoomlArt provides a plugin [T3 Framework] (http://www.t3-framework.org/) that includes those scripts and stylesheets to the document.
I guess, it may be an issue if FOF30 creates a cache file before plugins like T3 Framework includes the stylesheets to the header of the document.

If you would like, you can [download T3 Framework] (http://www.t3-framework.org/downloads.html) and test the module Akeeba Subscriptions List over it.
I have tested it on "T3 BS3 Blank Template".

![breakpoint_1](https://cloud.githubusercontent.com/assets/1272206/11712153/d8ebe190-9f33-11e5-8c46-e0c608156b50.png)

![breakpoint_3](https://cloud.githubusercontent.com/assets/1272206/11712156/de25124e-9f33-11e5-8b93-068513e39b4d.png)

![berakpoint_4](https://cloud.githubusercontent.com/assets/1272206/11712159/e1b05716-9f33-11e5-9f3b-4451a6c03bdd.png)

**Comment by nikosdion** [comment]
No, FOF 3.0 does NOT handle full page caching. It's Joomla! that does this. All FOF 3.0 does is ask JCache "hey there, is there's a cached version of this view?". Apparently something in T3 doesn't play nicely with Joomla? I don't know. It's not my code and I won't waste a day or ten debugging third party code. Especially not for a product that's clearly labelled "developer's product" and the condition of filing an issue is that you have debugged the problem and can provide a code fix.

The reason you haven't run into this issue before is that pretty much no other module out there is loading a component view. Therefore the JCache code is never executed and T3's bug isn't triggered.

Basically, you're barking at the wrong tree. FOF 3.0 doesn't do anything funky. It uses the standard Joomla! API.
